### PR TITLE
remote-test: use u64 to represent file size

### PR DESCRIPTION
**Pull request by onur-ozkan**
Currently, triggering a transfer of data exceeding the size of 4294967295 bytes results in a panic on the `remote-test-server` as `io::copy(&mut file, dst) failed with Connection reset by peer (os error 104)`. This issue happens because the size is transmitted as u32 to `remote-test-server`.

First commit increases the supported file size. But I am not sure about its necessity — can we realistically encounter file sizes exceeding 4GB in builds, perhaps through some complicated configurations?
 
~The second commit adds a sanity check to avoid encountering the error `io::copy(&mut file, dst) failed with Connection reset by peer (os error 104)` on the `remote-test-server` side.~

### CODE REVIEW

**Comment by onur-ozkan** [comment]
> First commit increases the supported file size. But I am not sure about its necessity — can we realistically encounter file sizes exceeding 4GB in builds, perhaps through some complicated configurations?

Let me know if this isn't needed (so I can revert).

**Comment by Mark-Simulacrum** [comment]
I don't think we need to support large files, though I also don't mind supporting them.

That said, if we're touching this code I think rather than manually bit twiddling using https://doc.rust-lang.org/nightly/std/primitive.u32.html#method.to_le_bytes and correspondign from_le_bytes feels much better to me.

**Comment by onur-ozkan** [comment]
> The second commit adds a sanity check to avoid encountering the error io::copy(&mut file, dst) failed with Connection reset by peer (os error 104) on the remote-test-server side.

Just realized that there is no need for a sanity check on the size anymore since we transmit the size without any conversion.

Updated the manual bit-twiddling part to use `to_be_bytes` on the client side (server side already uses `from_be_bytes`).

@rustbot ready

File: src/tools/remote-test-client/src/main.rs [hunk 1]
```diff
@@ -300,18 +300,18 @@ fn run(support_lib_count: usize, exe: String, all_args: Vec<String>) {
...
     while !stdout_done || !stderr_done {
         t!(client.read_exact(&mut header));
-        let amt = ((header[1] as u64) << 24)
-            | ((header[2] as u64) << 16)
-            | ((header[3] as u64) << 8)
-            | ((header[4] as u64) << 0);
+        let mut amt = 0u64;
+        for (i, &byte) in header[1..=8].iter().enumerate() {
+            amt |= (byte as u64) << (56 - 8 * i);
+        }
```
Comment (Mark-Simulacrum) [inline src/tools/remote-test-client/src/main.rs:314]: `u64::from_be_bytes(header[1..9].try_into().unwrap())`

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]

**Pull request merged**